application/src/modules/homepage/HomepageState.js

```diff
@@ -55,6 +66,7 @@ export default function HomepageStateReducer(state = initialState, action = {})
     case NOTIFICATION_RESPONSE:
       // We got a notification update so let's update the state and then restart the timer.
       return loop(
+        // TODO(@iprunache) stop triggering a render for every poll when the payload does not change; happens with immutable too.
```

Comment:
I think the article you mention describes the strategy for rendering after a render was decided.